### ISSUE TITLE
Database dropped even with PS_ERASE_DB=0

### ISSUE THREAD

**Issue by Tal-telecomsante** [PrestaShop/docker#103]
Hi,

Database is still dropped even if I use PS_ERASE_DB=0 (with the PS_INSTALL_AUTO=1)
Is it normal ? 

logs : 
* Reapplying PrestaShop files for enabled volumes ...

* Renaming admin folder as admintest ...

* Installing PrestaShop, this may take a while ...
Database "prestashop" dropped

Thank you !

**Comment by Quetzacoalt91** [comment]
Nope, this is unexpected. Have you been using this image for a long time? You may need to pull it again.

**Comment by Tal-telecomsante** [comment]
I just tried with the latest image, still have the problem.
No more 'Database dropped' logs, but my orders and customers are no more existing after removing container and create new one.

Here is my docker-compose

services:
  mysql-shop:
    image: mysql:latest
    container_name: mysql-shop
    restart: unless-stopped
    networks:
      telecomsante.loc:
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=true
      - MYSQL_DATABASE=prestashop
      - MYSQL_USER=prestashop
      - MYSQL_PASSWORD=prestashop

  shop:
    image: prestashop/prestashop:latest
    container_name: shop
    restart: unless-stopped
    networks:
      telecomsante.loc:
    expose:
      - 8080
    environment:
      - DB_SERVER=mysql-shop
      - DB_USER=prestashop
      - DB_PASSWD=prestashop
      - PS_INSTALL_AUTO=1
      - PS_ERASE_DB=0
      - PS_FOLDER_ADMIN=admin11
      - PS_LANGUAGE=fr
      - PS_COUNTRY=fr
      - ADMIN_MAIL=test@test.fr
      - ADMIN_PASSWD=test

**Comment by Quetzacoalt91** [comment]
Interesting, this could be cause by the file check.
https://github.com/PrestaShop/docker/blob/2d27d90d1b083da4f416db39860d879e581afeda/config_files/docker_run.sh#L9

When you delete and recreate a prestashop container without volume, this file does not exist anymore, which could make the installation process happen again.

Can you please add a volume in your docker-compose file for the folder /var/www/html? This seems to be a important missing point.

```docker-compose
shop:
  image: prestashop/prestashop:latest
  container_name: shop
  restart: unless-stopped
  networks:
      default.loc:
  expose:
      - 8080
  environment:
      - DB_SERVER=mysql-shop
      - DB_USER=prestashop
      - DB_PASSWD=prestashop
      - PS_INSTALL_AUTO=1
      - PS_ERASE_DB=0
      - PS_FOLDER_ADMIN=admin11
      - PS_LANGUAGE=fr
      - PS_COUNTRY=fr
      - ADMIN_MAIL=test@test.fr
      - ADMIN_PASSWD=test
    volumes:
      - data-volume:/var/www/html

volumes:
  data-volume:
```

**Comment by Tal-telecomsante** [comment]
Yes it's working now.

Thank you for help, I'm used to work with little services quite independent to work without data persistence app side :)

**Comment by Quetzacoalt91** [comment]
Glad I helped, you're welcome. :)

**Comment by Tal-telecomsante** [comment]
Hi,

I have a question for you !

Even if the docker_run script doesn't recognize the settings.inc.php file ( `
if [ ! -f ./config/settings.inc.php ]; `) it shouldn't drop the database, right ?

`if [ $PS_ERASE_DB = 1 ]; then`

This line is in the precedent 'if' block, with the settings.inc.php file check.

I'm asking it cause I want to run PrestaShop for first time with an already set database (with a sql dump)

Thank you :)

**Comment by Quetzacoalt91** [comment]
By default the database is not dropped. See https://github.com/PrestaShop/docker/blob/2d27d90d1b083da4f416db39860d879e581afeda/Dockerfile.model#L16

And even if you request it, it will be erased only if the file ./config/settings.inc.php is not found AND only if you asked for the automatic installation.

**Comment by Tal-telecomsante** [comment]
That's what I understand by checking the script, however the database is still dropped even with PS_ERASE_DB=0

I will retry tomorrow but that was the problem I describe in my first post : it's seems that without config file ( first installation ), PS_INSTALL_AUTO=1, PS_ERASE_DB=0 and a already full database, this one is dropped during installation.

**Comment by Quetzacoalt91** [comment]
Yes, that's quite unexpected. I'll need more details on the environment variable you use in your docker-composer OR in your `docker run` commands

**Comment by Tal-telecomsante** [comment]
Here is my docker-compose

`version: '2'

services:
  mysql-shop:
    image: mysql:latest
    container_name: mysql-shop
    restart: unless-stopped
    networks:
      test.loc:
    environment:
      - MYSQL_RANDOM_ROOT_PASSWORD=true
      - MYSQL_DATABASE=prestashop
      - MYSQL_USER=prestashop
      - MYSQL_PASSWORD=prestashop

  shop:
    image: prestashop/prestashop:1.7.1.2
    container_name: shop
    restart: unless-stopped
    networks:
      test.loc:
    environment:
      - PS_DOMAIN=shop.test.loc
      - DB_SERVER=mysql-shop
      - DB_USER=prestashop
      - DB_PASSWD=prestashop
      - PS_INSTALL_AUTO=1
      - PS_ERASE_DB=0
      - PS_FOLDER_ADMIN=myadminfolder
      - PS_LANGUAGE=fr
      - PS_COUNTRY=fr
      - ADMIN_MAIL=mymail
      - ADMIN_PASSWD=mypswd
    volumes:
      - prestashop-html:/var/www/html

volumes:
  prestashop-html:

networks:
  test.loc:
    external: true
`

**Comment by timactive** [comment]
We have same issue, i actually use this image in kubernetes cluster.
My env variable is same : PS_ERASE_DB=0 and PS_INSTALL_AUTO=1
And database is drop when i reinstall POD.
I mount only the persist folder /var/www/modules,/var/www/override, /var/www/themes, /var/www/img.

All time the script call index_cli.php, but i thinked it is normal if upgrade is need. but this script drop also the database.
php /var/www/html/$PS_FOLDER_INSTALL/index_cli.php

It is normal?

**Comment by Quetzacoalt91** [comment]
Can you please add a volume for the config folder @timactive?

Should be `app/config` on PS 1.7 and `config` on prior versions

**Comment by Quetzacoalt91** [comment]
Can you please add a volume for the `config` folder @timactive?

**Comment by timactive** [comment]
@Quetzacoalt91 
Thank, i understand that will resolve  but i request on this issue to more understand the logic, how working upgrade? we need to launch module upgrade manually?

**Comment by vicdup** [comment]
This is an unresolved issue. We are stuck on this, if you want to use Prestashop in a stateless mode (think Kubernetes), each time a pod start the DB is erased...

**Comment by PierreRambaud** [comment]
It should be ok for months :thinking: 
We are using PS_ERASE_DB=0 and PS_INSTALL_DB=0 and PS_INSTALL_AUTO=0.

**Comment by comur** [comment]
Hi,

This isn't fixed for me. I have the exact same issue, do you have a workaround. We deploy multiple prestashops for module development purposes and need to keep data on them but don't want to keep code except modules.

Do you have any workaround on this issue ?

Thx
